### ISSUE TITLE
[HierarchyList] accessibility errors

### ISSUE THREAD

**Issue by bjornalm** [carbon-design-system/carbon-addons-iot-react#3079]
### What package is this for?

- [x] React
- [ ] Angular

### Describe the bug

The HierarchyList has a couple of accessibility errors when using nested items and Editing Style is 'multiple-nesting'.

![image](https://user-images.githubusercontent.com/5167091/142629668-f26f7af4-b018-4c0a-ab7a-e65b11fad163.png)

### To Reproduce

1. Go to https://next.carbon-addons-iot-react.com/?path=/story/1-watson-iot-hierarchylist--with-nested-reorder
2. Enable  'multiple-nesting' in knob Editing Style 
2. Run the accessibility test
3. See error

### Expected behavior

There should be no accessibility errors

### Environment/versions:

- OS: [e.g. MacOS, Windows]
- Browser: [e.g. chrome, safari]
- `carbon-addons-iot-react` version: [e.g. `v2.60.0`]

### Specific timeline issues / requests

Do you want this work within a specific time period? Is it related to an
upcoming release?

_NB: The core contributors will try to work with your timeline, but it's not
guaranteed. The earlier you make a request in advance of a desired delivery
date, the better!_
